unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -2900,64 +2936,69 @@ public struct WhatDidITouch {
             }
 
-            if (CheckIfAgentCanRotateHand()) {
-                Vector3 vec = new Vector3(action.x, action.y, action.z);
-                AgentHand.transform.localRotation = Quaternion.Euler(vec);
-                // SetUpRotationBoxChecks();
+            if (!CheckIfAgentCanRotateHand()) {
+                throw new InvalidOperationException("Object is unable to rotate in its current state!");
+            }
 
-                // if this is rotated too much, drop any contained object if held item is a receptacle
-                if (Vector3.Angle(ItemInHand.transform.up, Vector3.up) > 95) {
-                    DropContainedObjects(
-                        target: ItemInHand.GetComponent<SimObjPhysics>(),
-                        reparentContainedObjects: true,
-                        forceKinematic: false
-                    );
-                }
+            AgentHand.transform.localRotation = Quaternion.Euler(rotation);
+            // SetUpRotationBoxChecks();
 
-                actionFinished(true);
-            } else {
-                actionFinished(false);
+            // if this is rotated too much, drop any contained object if held item is a receptacle
+            if (Vector3.Angle(ItemInHand.transform.up, Vector3.up) > 95) {
+                DropContainedObjects(
+                    target: ItemInHand.GetComponent<SimObjPhysics>(),
+                    reparentContainedObjects: true,
+                    forceKinematic: false
+                );
             }
+
+            actionFinished(true);
         }
 
         // rotate the hand if there is an object in it
-        public void RotateHandRelative(ServerAction action) {
-
+        public void RotateHeldObject(float pitch = 0, float yaw = 0, float roll = 0) {
             if (ItemInHand == null) {
-                errorMessage = "Can't rotate hand unless holding object";
-                return;
+                throw new InvalidOperationException("Can't rotate hand unless holding object.");
             }
 
             Quaternion agentRot = transform.rotation;
             Quaternion agentHandStartRot = AgentHand.transform.rotation;
 
             transform.rotation = Quaternion.identity;
 
+            // NOTE: -roll is used so that rotating an object rightward is positive
             AgentHand.transform.Rotate(
-                new Vector3(action.x, action.y, action.z), Space.World
+                new Vector3(pitch, yaw, -roll), Space.World
             );
             transform.rotation = agentRot;
 
             if (isHandObjectColliding(true)) {
-                errorMessage = "Hand object is coliding after rotation.";
                 AgentHand.transform.rotation = agentHandStartRot;
-                actionFinished(false);
-            } else {
-                actionFinished(true);
+                throw new InvalidOperationException("Hand object is coliding after rotation.");
```

Comment:
Also this. Perhaps update to "Held object is colliding after rotation."

Reply:
Updated in https://github.com/allenai/ai2thor/pull/770/commits/dcc9ef823c6793f49dba3c0b7a2f1027f5096a1b.